contracts/interfaces/IBondFactory.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.9;
+
+interface IBondFactory {
+    /**
+        @notice Emitted when the restriction of bond creation to allow-listed
+            accounts is toggled on or off.
+        @param isIssuerAllowListEnabled The new state of the allow list.
+    */
+    event IssuerAllowListEnabled(bool isIssuerAllowListEnabled);
+
+    /**
+        @notice Emitted when the restriction of collateralToken and paymentToken
+            to allow-listed tokens is toggled on or off.
+        @param isTokenAllowListEnabled The new state of the allow list.
+    */
+    event TokenAllowListEnabled(bool isTokenAllowListEnabled);
+
+    /**
+        @notice Emitted when a new bond is created.
+        @param newBond The address of the newley deployed bond.
+        @param name Passed into the ERC20 token to define the name.
+        @param symbol Passed into the ERC20 token to define the symbol.
+        @param owner Ownership of the created Bond is transferred to this
+            address by way of _transfeOwnership and tokens are minted to this address. See
+            `initialize` in `Bond`.
+        @param maturityDate The timestamp at which the Bond will mature.
+        @param paymentToken The ERC20 token address the Bond is redeemable for.
+        @param collateralToken The ERC20 token address the Bond is backed by.
+        @param collateralTokenAmount The amount of collateral tokens per bond.
+        @param convertibleTokenAmount The amount of convertible tokens per bond.
+        @param bonds The amount of Bonds given to the owner during the one-time
+            mint during the `Bond`'s `initialize`.
+    */
+    event BondCreated(
+        address newBond,
```

Comment:
<img width="304" alt="image" src="https://user-images.githubusercontent.com/15036618/163479484-c40937c2-c3d8-4527-8b22-5fe0fa64b920.png">
 
 > You can add the attribute indexed to up to three parameters which adds them to a special data structure known as [“topics”](https://docs.soliditylang.org/en/latest/abi-spec.html#abi-events) instead of the data part of the log. A topic can only hold a single word (32 bytes) so if you use a [reference type](https://docs.soliditylang.org/en/latest/types.html#reference-types) for an indexed argument, the Keccak-256 hash of the value is stored as a topic instead. 
 
 Interesting. Going to leave unindexed.